api/v1/installation_types.go

```diff
@@ -368,6 +368,11 @@ type NodeAddressAutodetection struct {
 	// +optional
 	FirstFound *bool `json:"firstFound,omitempty"`
 
+	// Kubernetes configures Calico to detect node addresses based on the Kubernetes API.
```

Comment:
@tmjd @Brian-McM what do you think about this API? Please give it a critical eye!

The reason behind not having it as a bool is that we might want to extend this to other options in the future. For example. any of these: https://kubernetes.io/docs/concepts/architecture/nodes/#addresses

Or our own modification - e.g., `NodeCalicoIP` - which looks for an address in the list with type `CalicoIP`

Reply:
I'm always a fan of not using a bool in the API, I think that's a good choice.
`kubernetes` seems very broad/generic but I think the possible values (and the comments on them) should clarify any ambiguity.